File: bubble_sort/bubble_sort.c
#include <stdio.h>

void BubbleSort(int *arr, int len)
{
    int temp;
    for (int i = 0; i < len - 1; i++)
    {
        for (int j = 0; j < len - 1 - i; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main()
{
    int arr[] = {1, 3, 2, 5, 8, 3, 9, 4};
    BubbleSort(arr, 8);
    for (int i = 0; i < 8; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
    return 0;
}

File: bubble_sort/bubble_sort.go
package main

import (
	"fmt"
)

func BubbleSort(arr []int) {
	var temp int
	for i := 0; i < len(arr)-1; i++ {
		for j := 0; j < len(arr)-1-i; j++ {
			if arr[j] > arr[j+1] {
				temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			}
		}
	}
}

func main() {
	arr := []int{2, 3, 6, 1, 5, 3, 9, 4}
	BubbleSort(arr)
	fmt.Println(arr)
}
